spring-kafka/src/main/java/org/springframework/kafka/retrytopic/ListenerContainerFactoryConfigurer.java

```diff
@@ -98,6 +98,7 @@ public class ListenerContainerFactoryConfigurer {
 
 	private final Clock clock;
 
+	@SuppressWarnings("deprecation")
```

Comment:
I don't see what is deprecated for this ctor to use...
Do we really need this annotation over here?

Reply:
Same, deprecated RetryTopicInternalBeanNames.

Reply:
OK. But why do you still use that deprecated class then?
Looks like those are just string constants, so why just don't use their values in place instead of referring to that deprecated class? That's for the case if we cannot get rid of these deprecated names yet...

Reply:
Well, it's a matter of how rigorous we need to be with backwards compatibility.

All current documentation points to users declaring beans with these names, and more curious users might have other creative solutions that rely on overriding some of these.

So my thinking is, for 2.9, anything that worked before, should still work just the same - that's why we should be able to introduce new names but still make sure old ones work. And for 3.0 we could get rid of the old ones. Is that thinking right?

But I'm ok with ripping the bandaid sooner if you think that's ok.

Reply:
BTW we can push RC1 out a week or so if necessary, it is not tied to the Boot release train.

Reply:
Sorry M1.